src/App.tsx

```diff
@@ -9,52 +10,192 @@
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
 
-export const App = () => (
-  <main className="section">
-    <div className="container">
-      <div className="tile is-ancestor">
-        <div className="tile is-parent">
-          <div className="tile is-child box is-success">
-            <div className="block">
-              <UserSelector />
-            </div>
+import { User } from './types/User';
+import { Post } from './types/Post';
+import { getUsers } from './api/user';
+import { getPosts } from './api/post';
+import { addComment, deleteComment, getComments } from './api/comment';
+import { Comment } from './types/Comment';
+
+export const App = () => {
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [users, setUsers] = useState<User[]>([]);
+  const [isLoadingUsers, setIsLoadingUsers] = useState(false);
+  const [isLoadingPosts, setIsLoadingPosts] = useState(false);
+  const [isLoadingComments, setIsLoadingComments] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [isFormOpen, setIsFormOpen] = useState(false);
+
+  useEffect(() => {
+    const fetchUsers = async () => {
+      setIsLoadingUsers(true);
+      try {
+        const usersData = await getUsers();
+
+        setUsers(usersData);
+        setError(null);
+      } catch (err) {
+        setError('Failed to load users');
+      } finally {
+        setIsLoadingUsers(false);
+      }
+    };
+
+    fetchUsers();
+  }, []);
+
+  useEffect(() => {
+    const fetchPosts = async () => {
+      setIsLoadingPosts(true);
+
+      try {
+        if (selectedUser) {
+          const userPosts = await getPosts(selectedUser.id);
+
+          setPosts(userPosts);
+          setError(null);
+        }
+      } catch (err) {
+        setError('Failed to load posts');
+      } finally {
+        setIsLoadingPosts(false);
+      }
+    };
+
+    fetchPosts();
+  }, [selectedUser]);
 
-            <div className="block" data-cy="MainContent">
-              <p data-cy="NoSelectedUser">No user selected</p>
+  useEffect(() => {
+    const fetchComments = async () => {
+      setIsLoadingComments(true);
+      try {
+        if (selectedPost) {
+          const postComments = await getComments(selectedPost.id);
 
-              <Loader />
+          setComments(postComments);
+          setError(null);
+        }
+      } catch (err) {
+        setError('Failed to load comments');
+      } finally {
+        setIsLoadingComments(false);
+      }
+    };
 
-              <div
-                className="notification is-danger"
-                data-cy="PostsLoadingError"
-              >
-                Something went wrong!
+    fetchComments();
+  }, [selectedPost]);
```

Comment:
Implement a 300 ms delay for API requests to align with the task requirements. This can be achieved using a `setTimeout` to delay the execution of the fetch functions.